snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -126,24 +126,27 @@ def __init__(
 
     def get_pull_state(self) -> states.PullState:
         if not self._pull_state:
-            self._pull_state = states.get_state(self.plugin.statedir, steps.PULL)
+            self._pull_state = cast(states.PullState, self.get_state(steps.PULL))
```

Comment:
why is this necessary?

Reply:
Because this PR adds static checks to the return type of `get_state`. This now means casting if it's a subclass.